geolocation/src/definitions.ts

```diff
@@ -0,0 +1,137 @@
+declare module '@capacitor/core' {
+  interface PluginRegistry {
+    Geolocation: GeolocationPlugin;
+  }
+}
+
+export type CallbackID = string;
+
+export type GeolocationPermissionType = 'location';
+export type GeolocationState = PermissionState;
+
+export interface GeolocationPermissionStatus {
+  location: GeolocationState;
+}
+
+export interface GeolocationPlugin {
+  /**
+   * Get the current GPS location of the device
+   *
+   * @since 1.0.0
+   */
+  getCurrentPosition(
+    options?: GeolocationOptions,
+  ): Promise<GeolocationPosition>;
+
+  /**
+   * Set up a watch for location changes. Note that watching for location changes
+   * can consume a large amount of energy. Be smart about listening only when you need to.
+   *
+   * @since 1.0.0
+   */
+  watchPosition(
+    options: GeolocationOptions,
+    callback: GeolocationWatchCallback,
+  ): CallbackID;
+
+  /**
+   * Clear a given watch
+   */
+  clearWatch(options: { id: string }): Promise<void>;
+
+  /**
+   * Check location permissions
+   *
+   * @since 1.0.0
+   */
+  checkPermissions(): Promise<GeolocationPermissionStatus>;
+
+  /**
+   * Request location permissions
+   *
+   * @since 1.0.0
+   */
+  requestPermissions(
+    permissions: GeolocationPermissionType,
+  ): Promise<GeolocationPermissionStatus>;
+}
+
+export interface GeolocationPosition {
+  /**
+   * Creation timestamp for coords
+   *
+   * @since 1.0.0
+   */
+  timestamp: number;
+
+  /**
+   * The GPS coordinates along with the accuracy of the data
+   *
+   * @since 1.0.0
+   */
+  coords: {
+    /**
+     * Latitude in decimal degrees
+     *
+     * @since 1.0.0
+     */
+    latitude: number;
+
+    /**
+     * longitude in decimal degrees
+     *
+     * @since 1.0.0
+     */
+    longitude: number;
+
+    /**
+     * Accuracy level of the latitude and longitude coordinates in meters
+     *
+     * @since 1.0.0
+     */
+    accuracy: number;
+
+    /**
+     * Accuracy level of the altitude coordinate in meters (if available)
+     *
```

Comment:
We can be more specific here

```suggestion
     * Accuracy level of the altitude coordinate in meters, if available.
     *
     * Available on all iOS versions and on Android 8.0+.
     *
```